server/SchemaLoader.js

```diff
@@ -0,0 +1,142 @@
+"use strict";
+const EventEmitter = require("events");
+const { remote } = require("electron");
+const logger = require("electron-log");
+const jsonfile = require("jsonfile");
+const mkdirp = require("mkdirp");
+const path = require("path");
+const fs = require("fs");
+
+const ValidationSchema = require("./model/ValidationSchema");
+
+class SchemaLoader extends EventEmitter {
+    constructor() {
+        super();
+
+        this.on("schemas-synced", function() {
+            this.loadLocalSchemas();
+        });
+    }
+
+    syncSchemas() {
+        let repos = remote.getGlobal("options").repos;
+
+        if (!!repos) {
+            repos.forEach((repo) => {
+                if (!!repo.url && !!repo.apikey && !!repo.vendors && repo.vendors.length > 0) {
+                    logger.info("Syncing " + repo.vendors + " schemas from " + repo.url);
+
+                    const req = new XMLHttpRequest();
+                    let endpoint = "/api/schemas/" + repo.vendors.join("%2C");
+                    req.open("GET", repo.url + endpoint);
+                    req.setRequestHeader("apikey", repo.apikey);
+                    req.onreadystatechange = function() {
+                        if (req.readyState === XMLHttpRequest.DONE) {
+                            // Request was sent, response is downloaded and operation is complete.
+                            if (req.status === 200) {
+                                this.storeSchemas(JSON.parse(req.responseText));
+                            } else if (req.status > 0) {
```

Comment:
> Even if there's nothing to store, such as 204 and 206?
Sure. 204 seems fine (though it'd be very odd) and 206 won't happen anyway because we're not doing a range request.

> Think we should then also accept 304 as successful. Agreed?
The way the request is set up, I don't think the server can ever return this (because we're not providing sufficient info about the local version, if any.)